boards/beagle/beagley_ai/doc/index.rst

```diff
@@ -0,0 +1,142 @@
+.. _beagley_ai:
+
+BeagleY-AI
+##########
+
+Overview
+********
+
+BeagleY-AI is a computational platform powered by TI AM67A (J722S) SoC, which is
+targeted for automotive applications.
+
+.. figure:: assets/byai.webp
+   :align: center
+   :width: 600px
+   :alt: BeagleBoard.org BeagleY-AI
+
+Hardware
+********
+BeagleY-AI is powered by TI AM67A (J722S) SoC, which has two domains (Main,
+MCU). This document gives overview of Zephyr running on both Cortex R5.
```

Comment:
I'd agree it would have been nice for the datasheet and TRM to show the more conventional block diagram, and not just the marketing "Functional Block Diagram". The names "Device Management" and "Run-time Management" only seem to show up in that one block diagram, I'm not sure where that name came from either..

Note there are actually 3 R5s in that diagram, the two you listed and the "MCU Channel" one. These map to domains:

* Device Management: WKUP
* MCU Channel: MCU
* Run-time Management: MAIN

The WKUP domain R5 is the boot master and runs DM firmware, this one we do not yet support running Zephyr as that would require Zephyr to provide DM services (which could be done, but is not yet on the roadmap).

The difference between the MAIN and MCU R5s is that the MCU one runs in an FFI isolated area with direct access to several isolated peripherals. There are some things you can do with that around low-power and safety, but those features are not yet supported in this initial support RP. So the two are basically identical from a software perspective at the moment.

Will update the docs as the differentiating support is enabled.